include/ck/utility/math.hpp

```diff
@@ -142,6 +142,22 @@ __host__ __device__ constexpr auto min(X x, Ys... ys)
     return min(x, min(ys...));
 }
 
+// disallow implicit type casting
+template <typename T>
+__device__ T exp(T x);
+
+template <>
+__device__ float exp<float>(float x)
+{
+    return __expf(x);
+}
+
+template <>
+__device__ double exp<double>(double x)
+{
+    return exp(x);
+}
+
```

Comment:
Here `exp` is not  consistent with other Unary element-wise operator, which uses functor and defines both output and input as operator parameters like `void operator()(T &y, const T&x)`.

Reply:
It's not intended to satisfy `tensor_operation::element_wise` constraints. It's just a free function ready to be called anywhere